### PR TITLE
Fix batch arrays

### PR DESCRIPTION
**Pull request by dirkkul**

### CODE REVIEW

File: weaviate/collection/data.py [hunk 1]
```diff
@@ -320,13 +327,35 @@ def __parse_properties_grpc(data: Dict[str, Any]) -> weaviate_pb2.BatchObject.Pr
...
+            elif isinstance(val, list) and isinstance(val[0], bool):
+                bool_arrays.append(weaviate_pb2.BooleanArrayProperties(key=key, vals=val))
+            elif isinstance(val, list) and isinstance(val[0], str):
+                text_arrays.append(weaviate_pb2.TextArrayProperties(key=key, vals=val))
+            elif isinstance(val, list) and isinstance(val[0], datetime.datetime):
+                text_arrays.append(
+                    weaviate_pb2.TextArrayProperties(
+                        key=key, vals=[self.__datetime_to_string(x) for x in val]
+                    )
+                )
```
Comment (tsmith023) [inline weaviate/collection/data.py:339]: Will the gRPC handler here be changed in the future to specific `DateArrayProperties` or is there no different between a date and a string across the network?

File: weaviate/collection/data.py [hunk 2]
```diff
@@ -320,13 +327,35 @@ def __parse_properties_grpc(data: Dict[str, Any]) -> weaviate_pb2.BatchObject.Pr
...
+            elif isinstance(val, list) and isinstance(val[0], uuid_package.UUID):
+                text_arrays.append(
+                    weaviate_pb2.TextArrayProperties(key=key, vals=[str(x) for x in val])
+                )
```
Comment (tsmith023) [inline weaviate/collection/data.py:343]: Likewise here for `UUIDArrayProperties` as above for dates

**Review by tsmith023 — APPROVED** [review]

File: weaviate/collection/data.py (hunk 1 above)
Comment (dirkkul) [reply weaviate/collection/data.py:339]: there is no difference, weavite figures it out based on the datatype in the schema

File: weaviate/collection/data.py (hunk 2 above)
Comment (dirkkul) [reply weaviate/collection/data.py:343]: same

**Review by tsmith023 — APPROVED** [review]

**Pull request merged**